header.c

```diff
@@ -2170,6 +2181,12 @@ int sam_hdr_add_pg(sam_hdr_t *bh, const char *name, ...) {
         hrecs = bh->hrecs;
     }
 
+    bh->hrecs->pgs_changed = 1;
+    if (sam_hdr_link_pg(bh) < 0) {
```

Comment:
This makes `sam_hdr_add_pg()` fail if an existing chain has a `@PG PP:` tag that points to a non-existent `@PG ID:`, which is a change from previous behaviour.  Is this something we actually want to fail on, or should we just ignore it (maybe by not setting `ret` to -1 when the [hash table lookup fails](https://github.com/samtools/htslib/blob/8859b09/header.c#L2081))?

I also see the `sam_hdr_link_pg()` [call](https://github.com/samtools/htslib/blob/8859b09/header.c#L1248) in `sam_hdr_rebuild()` didn't check the return value, which is why it used to work.

Reply:
> 
> 
> This makes `sam_hdr_add_pg()` fail if an existing chain has a `@PG PP:` tag that points to a non-existent `@PG ID:`, which is a change from previous behaviour. Is this something we actually want to fail on, or should we just ignore it (maybe by not setting `ret` to -1 when the [hash table lookup fails](https://github.com/samtools/htslib/blob/8859b09/header.c#L2081))?

Print a warning and continue, perhaps.

Reply:
Yes, it already seems to do the continue bit.